java/maven/src/org/netbeans/modules/maven/newproject/idenative/IDENativeTemplateHandler.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.maven.newproject.idenative;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import org.netbeans.api.progress.ProgressHandle;
+import org.netbeans.api.project.ProjectManager;
+import org.netbeans.api.templates.CreateDescriptor;
+import org.netbeans.api.templates.CreateFromTemplateHandler;
+import org.netbeans.api.templates.FileBuilder;
+import org.netbeans.modules.maven.api.archetype.ProjectInfo;
+import org.netbeans.modules.maven.newproject.TemplateUtils;
+import org.netbeans.modules.maven.spi.newproject.CreateProjectBuilder;
+import org.netbeans.spi.project.ui.support.CommonProjectActions;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.util.BaseUtilities;
+import org.openide.util.NbBundle;
+
+/**
+ * Allows to create the 'ide native' projects in the headless mode. Is not registered as an interceptor, but rather
+ * listed explicitly for a template it handles.
+ * @author sdedic
+ */
+public class IDENativeTemplateHandler extends CreateFromTemplateHandler {
```

Comment:
Can we use some other name than IDE native? There is no native code and I find the name confusing.